.ci/breaking-change-detector/rules/rules_resource_schema_test.go

```diff
@@ -0,0 +1,89 @@
+package rules
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func TestResourceSchemaRule_RemovingAField(t *testing.T) {
+	for _, tc := range resourceSchemaRule_RemovingAField_TestCases {
+		tc.check(resourceSchemaRule_RemovingAField, t)
+	}
+}
+
+type resourceSchemaTestCase struct {
+	name               string
+	oldResourceSchema  map[string]*schema.Schema
+	newResourceSchema  map[string]*schema.Schema
+	expectedViolations int
+}
+
+var resourceSchemaRule_RemovingAField_TestCases = []resourceSchemaTestCase{
```

Comment:
Can you add a test that includes a nested field being removed or renamed?

Reply:
This rule expects the schema to be flattened. I can totally add a nested field test to the schema test suite though.

Reply:
added to comparison_test